src/map/script.c

```diff
@@ -13356,11 +13388,15 @@ BUILDIN_FUNC(warppartner)
 
 	mapindex = mapindex_name2id(str);
 	if (mapindex) {
-		pc_setpos(p_sd,mapindex,x,y,CLR_OUTSIGHT);
-		script_pushint(st,1);
+		if (buildin_warp_sub(p_sd, str, x, y))
```

Comment:
Invert the check and remove the outer else this way the two pushint 0 can be merged.